### PR TITLE
const keyword: brief paragraph on 'const fn'

### PR DESCRIPTION
**Pull request by RalfJung**
`const fn` were mentioned in the title, but called "deterministic functions" which is not their main property (though at least currently it is a consequence of being const-evaluable). This adds a brief paragraph discussing them, also in the hopes of clarifying that they do *not* have any effect on run-time uses.

### CODE REVIEW

File: library/std/src/keyword_docs.rs [hunk 1]
```diff
@@ -102,7 +102,9 @@ mod break_keyword {}
 
 #[doc(keyword = "const")]
 //
-/// Compile-time constants and deterministic functions.
+/// Compile-time constants and compile-time evaluable functions.
```
Comment (jyn514) [inline library/std/src/keyword_docs.rs:105]: TIL 'evaluable' means 'able to be evaluated'! https://english.stackexchange.com/questions/304948/evaluable-vs-evaluatable

**Review by jyn514 — APPROVED** [review]

File: library/std/src/keyword_docs.rs [hunk 2]
```diff
@@ -145,15 +147,28 @@ mod break_keyword {}
...
 /// The `const` keyword is also used in raw pointers in combination with `mut`, as seen in `*const
 /// T` and `*mut T`. More about `const` as used in raw pointers can be read at the Rust docs for the [pointer primitive].
 ///
-/// For more detail on `const`, see the [Rust Book] or the [Reference].
-///
 /// [pointer primitive]: primitive.pointer.html
 /// [Rust Book]:
 /// ../book/ch03-01-variables-and-mutability.html#differences-between-variables-and-constants
 /// [Reference]: ../reference/items/constant-items.html
+/// [cosnt-eval]: ./reference/const_eval.html
```
Comment (darksv) [inline library/std/src/keyword_docs.rs:171]:
```suggestion
/// [const-eval]: ./reference/const_eval.html
```

Comment (RalfJung) [reply library/std/src/keyword_docs.rs:171]: *oops*

Strange that the link checker didn't catch this though

**Pull request merged**